pageserver/src/storage_sync2.rs

```diff
@@ -366,14 +370,19 @@ impl UploadQueue {
             }
         }
 
-        let mut files = HashMap::new();
-        for path in &index_part.timeline_layers {
+        let mut files = HashMap::with_capacity(index_part.timeline_layers.len());
+        let timeline_path = conf.timeline_path(&timeline_id, &tenant_id);
+        for timeline_name in &index_part.timeline_layers {
```

Comment:
Looks like something is wrong with naming. Should it be layer_file_name instead of timeline_name?

Reply:
IndexPart should already contain remote paths, why do we need to do some conversions here?

Reply:
> IndexPart should already contain remote paths, why do we need to do some conversions here?

Not anymore, they've got removed in this PR actually, to emphasize that we list only layer (file) names in the IndexPart.